CMakeLists.txt

```diff
@@ -0,0 +1,60 @@
+cmake_minimum_required(VERSION 3.20)
+
+project(BigBaseV2 CXX ASM_MASM)
+
+set(SRC_DIR "${PROJECT_SOURCE_DIR}/BigBaseV2/src")
+
+set(CMAKE_CONFIGURATION_TYPES Release)
+
+
+# Fetch content
+message("\nFetching modules")
+include(scripts/minhook.cmake)
+include(scripts/g3log.cmake)
+include(scripts/pugixml.cmake)
+include(scripts/json.cmake)
+
+message("\nFetching custom modules")
+include(scripts/imgui.cmake)
+include(scripts/gtav-classes.cmake)
+
+
+# BigBaseV2
+message(STATUS "BigBaseV2")
+file(GLOB_RECURSE SRC_MAIN
+    "${SRC_DIR}/**.hpp"
+    "${SRC_DIR}/**.h"
+    "${SRC_DIR}/**.cpp"
+    "${SRC_DIR}/**.cc"
+    "${SRC_DIR}/**.cxx"
+    "${SRC_DIR}/**.asm"
+)
+add_library(BigBaseV2 MODULE "${SRC_MAIN}")
+
+set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+set_property(TARGET BigBaseV2 PROPERTY CXX_STANDARD 23) # 23 Because std::format is not avalible in std:c++20 for some reason. Maybe it's because i use v142 toolset.
+
+source_group(TREE "${SRC_DIR}" PREFIX "YimMenu" FILES "${SRC_MAIN}" )
+
+target_include_directories(BigBaseV2 PRIVATE 
+    "${SRC_DIR}"
+    "${json_SOURCE_DIR}/single_include"
+    "${gtav_classes_SOURCE_DIR}"
+    "${minhook_SOURCE_DIR}/include"
+    "${imgui_SOURCE_DIR}"
+)
+
+target_precompile_headers(BigBaseV2 PRIVATE "${SRC_DIR}/common.hpp")
+target_link_libraries(BigBaseV2 PRIVATE pugixml minhook g3log imgui d3dcompiler)
+
+# Disable warnings
+set_target_properties(BigBaseV2 PROPERTIES COMPILE_FLAGS "/wd4996") # C4996: deprecated in C++17
```

Comment:
Where does this warning occur? Let's try to fix it instead of silencing it, we're on C++20.

Reply:
This warning doesn't occur anymore but did when i added this.